BeagleBone_black/src/navigation/main.cpp

```diff
@@ -41,6 +44,14 @@ void Main::run()
 
   *last_readings = data_.getSensorsData();  // TODO(Brano): Make sure data_ is properly initd
   while (1) {
+    // State updates
+    State current_state = data_.getStateMachineData().current_state;
+    if (current_state != State::kIdle
+        && current_state != State::kReady
+        && nav_.getState() != NavigationState::kOperational
+        && nav_.getState() != NavigationState::kCriticalFailure)
```

Comment:
I would advice specifying in which nav states the finishCalibration is possible, rather than in which it is not. e.g. have only `nav_.getState() == NavigationState::kReady` as the condition.